reconcile/test/test_utils_state.py

```diff
@@ -0,0 +1,65 @@
+import os
+from tempfile import NamedTemporaryFile
+
+import pytest
+import boto3
+from moto import mock_s3
+from reconcile.utils.state import State
+
+
+@pytest.fixture
+def accounts():
+    """Account name is needed to instantiate a State"""
+    return [{'name': 'some-account'}]
+
+
+@pytest.fixture
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ['AWS_ACCESS_KEY_ID'] = 'testing'
+    os.environ['AWS_SECRET_ACCESS_KEY'] = 'testing'
+    os.environ['AWS_SECURITY_TOKEN'] = 'testing'
+    os.environ['AWS_SESSION_TOKEN'] = 'testing'
+
+
+@mock_s3
+def test_ls(accounts, aws_credentials, mocker):
+    os.environ['APP_INTERFACE_STATE_BUCKET'] = 'some-bucket'
+    os.environ['APP_INTERFACE_STATE_BUCKET_ACCOUNT'] = 'some-account'
+    s3_client = boto3.client('s3', region_name='us-east-1')
+    s3_client.create_bucket(Bucket='some-bucket')
+
+    file_text = 'test'
+    with NamedTemporaryFile(delete=True, suffix='.txt') as tmp:
+        with open(tmp.name, 'w', encoding='UTF-8') as f:
+            f.write(file_text)
+
+        s3_client.upload_file(tmp.name, 'some-bucket',
```

Comment:
You might consider `put_object` instead of `upload_file` to avoid needing to deal with the filesystem at all.  You'll see this is also how `__setitem__` is implemented for this class.

Reply:
Done.